src/main/java/io/cdap/plugin/http/sink/batch/HTTPSink.java

```diff
@@ -73,13 +75,22 @@ public void prepareRun(BatchSinkContext context) {
       .setFqn(config.getUrl()).build();
     LineageRecorder lineageRecorder = new LineageRecorder(context, asset);
     lineageRecorder.createExternalDataset(context.getInputSchema());
-    List<String> fields = inputSchema == null ?
-      Collections.emptyList() :
-      inputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
+    List<String> fields;
+    if (inputSchema == null) {
+      fields = Collections.emptyList();
+    } else {
+      assert inputSchema.getFields() != null;
```

Comment:
are you catching the assertion error somewhere?
if not, you can go ahead with old format, CDF has some validations for schema which will check it.

Reply:
handled using java 8, removed assert